Fix /sbin/sfdisk hanging indefinitely

hankjacobs opened the pull request:
Due to a kernel bug
(https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=897572#82) reading
from /dev/urandom can block. `sfdisk` appears to read from /dev/urandom
and would hang indefinitely due to this bug. This workaround generates
entropy so crng init can complete and reading from /dev/urandom will
unblock.

hankjacobs commented:
I could be really off base with this PR but it's the only thing that I could find that explains and resolves `sfdisk` hanging when running the tests. I'm also assuming that `fmt.Print` is generating entropy though I'm 100% sure about that. The kernel is not my area of expertise. 

The reason I jumped to this bug is `sfdisk` would only hang briefly when I printed in a tight loop from another goroutine and then the following line would be printed:
```
[    1.727285] random: crng init done
```

bradfitz commented:
This seems a little weird. We have to spin the CPU & write to stdout to generate entropy? We can't do anything else? Maybe we just need to wire up a randomness device (virtio_rng or something?) via qemu or something?

hankjacobs committed:
Fix /sbin/sfdisk hanging indefinitely

Wire up virtio-rng. This fixes an issue where sfdisk would hang
indefinitely waiting on a read from urandom
(https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=897572#82) due to a
lack of entropy.

hankjacobs commented:
Good point. I've pushed up a change that does exactly that. It's definitely a better fix. Thanks pointing me in the right direction.

bradfitz commented:
Thanks!
